instrumentation/play/play-mvc/play-mvc-2.4/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/play/v2_4/ActionInstrumentation.java

```diff
@@ -69,21 +66,16 @@ public static void stopTraceOnResponse(
         @Advice.Thrown Throwable throwable,
         @Advice.Argument(0) Request<?> req,
         @Advice.Return(readOnly = false) Future<Result> responseFuture,
-        @Advice.Local("otelContext") Context context,
-        @Advice.Local("otelScope") Scope scope) {
-      if (scope == null) {
+        @Advice.Enter @Nullable AdviceScope actionScope) {
+      if (actionScope == null) {
         return;
       }
-      scope.close();
+      actionScope.closeScope();
+
+      updateSpan(actionScope.getContext(), req);
```

Comment:
we should be able to move those directly into `AdviceScope.end()` by adding `req` as parameter, given the `updateSpan` is only used here we can also inline it in `AdviceScope.end` which allows to remove `getContext` and `closeScope` methds as well.